### PR TITLE
Harvest: Use <Infos/> for error messages

### PR DESCRIPTION
**Pull request by CPatchane**
Display errors using the Infos component from CozyUI
Screenshot 2019-05-09 at 15 07 23

### CODE REVIEW

File: packages/cozy-harvest-lib/src/components/AccountForm/Error.jsx [hunk 1]
```diff
@@ -36,9 +36,9 @@ export class AccountFormError extends PureComponent {
    * @param  {Error} error      The error
    * @param  {Object} konnector konnector related to this error
    * @param  {Func} t           Translation function, expected to be Polyglot.t()
-   * @return {String}           The error description
+   * @return {Object}           The error locale object (title + description)
    */
-  getDescription() {
+  getErrorLocale() {
```
Comment (gregorylegarec) [inline packages/cozy-harvest-lib/src/components/AccountForm/Error.jsx]: I am not very comfortable with the closure decltared in the middle of the method.

I think I'll do something like:

```jsx

createSuffixApppender = suffixKey => key => `${key}.${suffixKey}`

[...]

getErrorLocale(suffixKey) {
  const { error, konnector, t } = this.props

  const args = {
    name: konnector.name || '',
    link: konnector.vendor_link || ''
  }

  const withSuffix = createSuffixApppender(suffixKey)

  if(error instanceof KonnectorJobError) {
    return t(withSuffix(error.code), {
      ...args,
      _: t(withSuffix(error.type), {
        ...args,
        _: t(withSuffix(this.defaultKey), args)
      })
    })
  }
}
```

And in render method:

```jsx
    // [...]
    return (
      <Infos
         title={this.getErrorLocale('title')}
         text={<Markdown source={this.getErrlrLocale('description')} />}
         isImportant
      />
   )
```

I find this much simpler and readable. What do you think ?

Comment (CPatchane) [reply packages/cozy-harvest-lib/src/components/AccountForm/Error.jsx]: It's like something I was thinking about but what do you do with this part? https://github.com/cozy/cozy-libs/blob/master/packages/cozy-harvest-lib/src/components/AccountForm/Error.jsx#L59-L60

Comment (gregorylegarec) [reply packages/cozy-harvest-lib/src/components/AccountForm/Error.jsx]: I guess something like:
```jsx
    // [...]
	const description = this.getErrlrLocale('description')
    return (
      <Infos
         title={this.getErrorLocale('title')}
         text={<Markdown source={error.message ? `${description} (${error.message})` : description
} />}
         isImportant
      />
   )
```

Comment (CPatchane) [reply packages/cozy-harvest-lib/src/components/AccountForm/Error.jsx]: Fixed

**Pull request merged**